keps/sig-release/3720-freezing-k8s-gcr-io/README.md

```diff
@@ -0,0 +1,172 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The Kubernetes project has created and runs the `registry.k8s.io` image registry which is fully controlled by the community.
+This registry has been [GA for several months](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) now and we need to freeze the old image registry.
+
+## Motivation
+
+Running public image registries is very expensive and eats up a significant chunk of the project's Infrastructure budget. We built `registry.k8s.io` image registry to serve images from various origins around the world depending on the location of the user. For example, an a kops Kubernetes cluster in eu-west-1 can pull images from an AWS S3 bucket in the same region which is very fast and more importantly very cheap for the Kubernetes project.
+
+There was a plan to redirect `k8s.gcr.io` to `registry.k8s.io` but it didn't [work out](https://kubernetes.slack.com/archives/CCK68P2Q2/p1666725317568709) so we backported the image registry defaults to 1.24, 1.23, and 1.22 so all the patch releases from December 2022 will using the new registry by default.
+
+We are currently exceeding our budget as it will take quite a while for end users to upgrade Kubernetes to v1.25 so we want to incentivise our end users to move to the new registry as fast as possible by freezing the registry by 1.27. This would mean that all subsequent image releases will not be available on the old registry.
+
+### Goals
+
+Freeze `k8s.gcr.io` image registry and push all new images and tags exclusively to the `registry.k8s.io` image registry.
+
+### Non-Goals
+
+- `registry.k8s.io` internal implementations details. That is handled separately by sig-k8s-infra.
+
+## Proposal
+
+Freeze the `k8s.gcr.io` image by not pushing any new digests or tags after 1.27 release. The 1.27 release itself won't be available on `k8s.gcr.io`.
+
+I'm proposing that on the 1st of April 2023 (10 days before 1.27 is released):
+
+- `k8s.gcr.io` is frozen and no new images will be published by any subproject.
+- last 1.23 release on `k8s.gcr.io` will be 1.23.18 (goes EoL before the freeze)
+- last 1.24 release on `k8s.gcr.io` will be 1.24.12
+- last 1.25 release on `k8s.gcr.io` will be 1.25.8
+- last 1.26 release on `k8s.gcr.io` will be 1.26.3
+- 1.27.0 will not be available `k8s.gcr.io`
+
+### Risks and Mitigations
+
+There are no risks. The old registry will still be available and you can pull the images before 1.27 on there. This change will also
+affect other users of k8s.gcr.io who should have already updated their helm charts and manifests to use the new registry.
+
+## Design Details
```

Comment:
one important detail from past discussion not captured here:

syncing layers to s3 for consumption by registry.k8s.io currently depends on rclone of the GCS bucket exposed by GCR

we need a private/non-documented/clearly-not-intended-for-end-users GCR to continue publishing to in the short term, just for syncing. 

in the long term we should pull the contents from OCI. I started a tool for this in december but got sidetracked 🙃, this does NOT need to happen as part of the KEP, but to unblock phasing out GCR immediately we will need the new private stop-gap GCR for syncing. the manifests promoting to AR can promote to this one.

Reply:
I forgot about that. I can add it to the list of things I need to do in February.

Reply:
Thanks!

Also another long term option: We could instead look again at s3 layer sync in the image promoter, but regardless of long term plans we'll want to just add a temp GCR so we can phase out the main GCRs ASAP and sort out alternatives later.

Reply:
The long term option may be viable sooner than later, I got back to hacking on this and it's relatively straight-forward to implement against AR/GCR thanks to crane/gcrane. https://github.com/kubernetes/registry.k8s.io/pull/151

Not production grade yet, but will keep iterating towards that in parallel to the rest of the freeze effort.